### DISCUSSION TITLE
Questions on Schema

### DISCUSSION THREAD

**Discussion by joelthe1** [BibleNLP/bible-graph#6] [General]
Here are a few questions that maybe worth discussion (from Kavitha):
1. Why not use BCP-47?
2. How do we support languages that don't yet have an official language code (i.e. `x-`)
3. BibleRevision is directly related to verseTexts. As they are not related to BookReferences , chapterReferences or verseReferences, I assume, the book-chapter-verse structure is to be common. Do we plan to store bible specific data in these nodes, like book names in those languages or chapter labels(\cl)?

**Comment by DCNemesis** [comment]
1. I'm not sure we want to store things with fully specified BCP-47, but I think the graph should contain everything necessary for someone to construct a BCP-47 identifier. So, if we've got ISO 639-3, we'd also need to link those back to ISO 639-2/1 (where possible) and include BCP-47 registered languages(?). We've got ISO 15924.  adding regions, variants, etc. would get us the rest of the way. (Also, I think if we fully support BCP-47, that also answers 2)

**Reply by joelthe1** [reply]
From my understanding BCP-47 is superset of a few other language code standards and can gracefully fallback to using one of these subsets when extra information is not present. Is this also your understanding?

**Reply by DCNemesis** [reply]
As you were commenting, I created [an Issue](https://github.com/BibleNLP/bible-graph/issues/16) for this and what we would need to do to support this more fully.

**Comment by DCNemesis** [comment]
On 3 ... I agree that localized book names, headers, etc. are important metadata. I think the problem is that some sources (like the JHU corpus) don't preserve this. Would a localized-book-name node that links back to the main reference and to a language (perhaps also to the version) be better than putting all of this as extra information in each node? I think that depends a lot on our primary intended use-case for this graph. I think for a lot of NLP/AI tasks, the book name information is not used, and adding it to the node would increase query overhead.

**Reply by DCNemesis** [reply]
Put in a [pull request](https://github.com/BibleNLP/bible-graph/pull/15) for the addition of localized book names. That seems easiest to add in, and as we have book names in English in the reference chain, it seems we should support local languages there.
